contracts/contracts/coordination/InfractionCollector.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import "./Coordinator.sol";
+import "../../threshold/ITACoChildApplication.sol";
+
+contract InfractionCollector is OwnableUpgradeable {
+    Coordinator public coordinator;
+
+    // Reference to the TACoChildApplication contract
+    ITACoChildApplication public tacoChildApplication;
+
+    // infraction types
+    enum InfractionType {
+        MISSING_TRANSCRIPT
+    }
+
+    // Mapping to keep track of reported infractions
+    mapping(uint32 => mapping(address => mapping(InfractionType => bool))) public infractions;
```

Comment:
What is the reason why the top-level index here is the ritual ID?  I'm not suggesting any changes at this moment but how this data structure is formed has implications on how cost-effective and code-efficient looking up infractions is later.

Reply:
> how this data structure is formed has implications on how cost-effective and code-efficient looking up infractions is later.

This is a really interesting consideration.

I don't have clear thoughts on this at the moment, but is it more likely that infractions would be looked up/checked via staking provider address instead of by ritual id? This would be from a dashboard or ...(?). 

For example, would we want to know all infractions for a specific staker address as opposed to all infractions from a failed ritual id? Even in the latter case (all infractions from failed ritual id), it is easy to get the list of all staking provider addresses for a ritual id from the coordinator (which this contract also has a reference to), and then check infractions for each staking provider address.